internals/secrethub/run.go

```diff
@@ -391,12 +410,16 @@ func (e EnvFile) Secrets() []string {
 
 // NewEnv loads an environment of key-value pairs from a string.
 // The format of the string can be `key: value` or `key=value` pairs.
-func NewEnv(r io.Reader, vars map[string]string) (EnvSource, error) {
-	env, parser, err := parseEnvironment(r)
+func NewEnv(r io.Reader, vars map[string]string, parser tpl.Parser) (EnvSource, error) {
+	env, defaultParser, err := parseEnvironment(r)
```

Comment:
Is there a reason for evaluating this if `parser==nil`?

Reply:
Yes, we still need `env` and `err` if `parser != nil`. If `parser == nil` we also need the `defaultParser`.